### PR TITLE
MINOR: Avoid trace logging computation in `checkEnoughReplicasReachOffset`

### PR DESCRIPTION
**Pull request by ijuma**
`numAcks` is only used in the `trace` logging statement so it should be a `def` instead of a `val`. Also took the chance to improve the code and documentation a little.

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @guozhangwang.

File: core/src/main/scala/kafka/cluster/Partition.scala [hunk 1]
```diff
@@ -306,17 +306,18 @@ class Partition(val topic: String,
       case Some(leaderReplica) =>
         // keep the current immutable replica list reference
         val curInSyncReplicas = inSyncReplicas
-        val numAcks = curInSyncReplicas.count(r => {
+
+        def numAcks = curInSyncReplicas.count { r =>
```
Comment (ijuma) [inline core/src/main/scala/kafka/cluster/Partition.scala:312]: @guozhangwang @junrao Is it correct that we don't use `numAcks` in this method outside of logging?

**Comment by ewencp** [comment]
@ijuma This looks good to me, though I wonder why `def` instead of `lazy val`?

**Comment by ijuma** [comment]
@ewencp lazy val is good when something is used more than once as there is some cost (double-checked locking with an additional field to capture whether initialisation has taken place). Since `numAcks` is used only once, a plain `def` (i.e. a method) seemed like the right option.

**Comment by guozhangwang** [comment]
LGTM. Please feel free to merge.

**Comment by ewencp** [comment]
LGTM

**Pull request closed without merging**